zio-json/shared/src/main/scala-3/zio/json/macros.scala

```diff
@@ -595,13 +622,43 @@ object DeriveJsonEncoder extends Derivation[JsonEncoder] { self =>
     }
 
   def split[A](ctx: SealedTrait[JsonEncoder, A]): JsonEncoder[A] = {
+    val isEnumeration = ctx.subtypes.forall(_.typeclass == caseObjectEncoder)
```

Comment:
I chose to use this hack because I don't think I have access to the `ctx.params` of the subtypes here anymore.